### PR TITLE
tool-web: use ProcessUtil.run to invoke child processes

### PR DESCRIPTION
**Pull request by kevmoo**
All invocations are logged on verbose
Better, more detailed exceptions are thrown on errors
A lot less boilerplate

### CODE REVIEW

**Review by yjbanov — APPROVED** [review]
lgtm

File: packages/flutter_tools/lib/src/build_system/targets/web.dart [hunk 1]
```diff
@@ -189,33 +178,35 @@ class Dart2JSTarget extends Dart2WebTarget {
...
+        outputJSFile.path,
+        environment.buildDir.childFile('app.dill').path, // dartfile
+      ],
+    );
+    final File dart2jsDeps = environment.buildDir.childFile('app.dill.deps');
     if (!dart2jsDeps.existsSync()) {
-      environment.logger.printWarning('Warning: dart2js did not produced expected deps list at '
-        '${dart2jsDeps.path}');
+      environment.logger.printWarning(
+        'Warning: dart2js did not produced expected deps list at '
```
Comment (hterkelsen) [inline packages/flutter_tools/lib/src/build_system/targets/web.dart:207]: nit: produced -> produce

**Review by harryterkelsen — APPROVED** [review]
LGTM with typo nit

File: packages/flutter_tools/lib/src/build_system/targets/web.dart (hunk 1 above)
Comment (kevmoo) [reply packages/flutter_tools/lib/src/build_system/targets/web.dart:207]: Ha! Uh. Exercise for later. This is just code reflow and not any change I made

Comment (Hixie) [reply packages/flutter_tools/lib/src/build_system/targets/web.dart:207]: we should fix it now, it's trivial to fix and makes the product better :-)

**Comment by Hixie** [comment]
test-exempt: code refactor with no semantic change

File: packages/flutter_tools/lib/src/build_system/targets/web.dart (hunk 1 above)
Comment (Hixie) [reply packages/flutter_tools/lib/src/build_system/targets/web.dart:207]: looks like this wasn't fixed in the version that landed?

Comment (kevmoo) [reply packages/flutter_tools/lib/src/build_system/targets/web.dart:207]: https://github.com/flutter/flutter/pull/126114

**Pull request merged**